commit 1d380067623559c94f916298df657de6d4d85fc9
Author: Susan Franklin <sfranklin>
Date:   Thu Jul 24 18:43:43 2008 +0000

    Bug 242007 - [ui] UI code should check CANCEL status and OperationCanceledException

1	1	bundles/org.eclipse.equinox.p2.ui.sdk/META-INF/MANIFEST.MF
2	0	bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/ProvSDKUIActivator.java
3	0	bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/updates/AutomaticUpdater.java
4	2	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/PlanStatusHelper.java
4	0	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/RevertProfileWizardPage.java
4	0	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/actions/ProfileModificationAction.java
diff --git a/bundles/org.eclipse.equinox.p2.ui.sdk/META-INF/MANIFEST.MF b/bundles/org.eclipse.equinox.p2.ui.sdk/META-INF/MANIFEST.MF
index d5954dd..13e76a0 100644
--- a/bundles/org.eclipse.equinox.p2.ui.sdk/META-INF/MANIFEST.MF
+++ b/bundles/org.eclipse.equinox.p2.ui.sdk/META-INF/MANIFEST.MF
@@ -4,7 +4,7 @@ Bundle-Name: %bundleName
 Bundle-Vendor: %providerName
 Bundle-Localization: plugin
 Bundle-SymbolicName: org.eclipse.equinox.p2.ui.sdk;singleton:=true
-Bundle-Version: 1.0.0.qualifier
+Bundle-Version: 1.0.100.qualifier
 Bundle-Activator: org.eclipse.equinox.internal.p2.ui.sdk.ProvSDKUIActivator
 Import-Package: org.eclipse.equinox.internal.p2.core.helpers,
  org.eclipse.equinox.internal.provisional.p2.artifact.repository,
diff --git a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/ProvSDKUIActivator.java b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/ProvSDKUIActivator.java
index 6f48c9f..b814a73 100644
--- a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/ProvSDKUIActivator.java
+++ b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/ProvSDKUIActivator.java
@@ -206,6 +206,8 @@ public class ProvSDKUIActivator extends AbstractUIPlugin {
 				public boolean continueWorkingWithPlan(ProvisioningPlan plan, Shell shell) {
 					if (plan == null)
 						return false;
+					if (plan.getStatus().getSeverity() == IStatus.CANCEL)
+						return false;
 					// If the plan requires install handler support, we want to open the old update UI
 					if (UpdateManagerCompatibility.requiresInstallHandlerSupport(plan)) {
 						MessageDialog dialog = new MessageDialog(shell, ProvSDKMessages.ProvSDKUIActivator_UnsupportedFeatureTitle, null, ProvSDKMessages.ProvSDKUIActivator_UnsupportedFeatureMessage, MessageDialog.WARNING, new String[] {ProvSDKMessages.ProvSDKUIActivator_LaunchUpdateManager, IDialogConstants.CANCEL_LABEL}, 0);
diff --git a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/updates/AutomaticUpdater.java b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/updates/AutomaticUpdater.java
index 80000dc..5ef8ab4 100644
--- a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/updates/AutomaticUpdater.java
+++ b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/updates/AutomaticUpdater.java
@@ -81,6 +81,9 @@ public class AutomaticUpdater implements IUpdateListener {
 					// We are just checking prefs to determine whether to to show the popup or not.
 					String openPlan = prefs.getString(PreferenceConstants.PREF_OPEN_WIZARD_ON_ERROR_PLAN);
 					if (plan != null) {
+						// If the user cancelled the operation, don't continue
+						if (plan.getStatus().getSeverity() == IStatus.CANCEL)
+							return false;
 						boolean noError = plan.getStatus().getSeverity() != IStatus.ERROR;
 						if (noError || !(MessageDialogWithToggle.NEVER.equals(openPlan))) {
 							// Show the affordance if user prefers always opening a currentPlan or being prompted
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/PlanStatusHelper.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/PlanStatusHelper.java
index 3bd9a17..2de8252 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/PlanStatusHelper.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/PlanStatusHelper.java
@@ -53,12 +53,14 @@ public class PlanStatusHelper {
 	public static IStatus computeStatus(ProvisioningPlan plan, IInstallableUnit[] ius) {
 		if (plan == null)
 			return getStatus(IStatusCodes.UNEXPECTED_NOTHING_TO_DO, null);
-		// If the plan is ok, nothing to do
-		if (plan.getStatus().isOK())
+		// If the plan is ok or cancel, no further analysis is needed
+		if (plan.getStatus().isOK() || plan.getStatus().getSeverity() == IStatus.CANCEL)
 			return plan.getStatus();
+
 		// If this is a status we have already checked, don't bother doing so again.
 		if (plan.getStatus().getCode() == IStatusCodes.UNEXPECTED_NOTHING_TO_DO)
 			return plan.getStatus();
+
 		// If the plan has no IU operands and some were expected, then nothing will happen.
 		if (ius != null && ius.length > 0) {
 			boolean iusInPlan = false;
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/RevertProfileWizardPage.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/RevertProfileWizardPage.java
index e9da4ae..4a026ce 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/RevertProfileWizardPage.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/RevertProfileWizardPage.java
@@ -12,6 +12,7 @@ package org.eclipse.equinox.internal.p2.ui.dialogs;
 
 import java.lang.reflect.InvocationTargetException;
 import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
 import org.eclipse.equinox.internal.p2.ui.ProvUIMessages;
 import org.eclipse.equinox.internal.p2.ui.viewers.IUDetailsLabelProvider;
 import org.eclipse.equinox.internal.provisional.p2.core.ProvisionException;
@@ -231,6 +232,9 @@ public class RevertProfileWizardPage extends WizardPage {
 					ProvisioningOperationRunner.run(op, getShell(), StatusManager.SHOW | StatusManager.LOG);
 					return true;
 				}
+				// If user cancelled, do not report an error
+				if (plan[0].getStatus().getSeverity() == IStatus.CANCEL)
+					return false;
 				ProvUI.reportStatus(plan[0].getStatus(), StatusManager.LOG);
 				setMessage(ProvUIMessages.ProfileModificationWizardPage_UnexpectedError, IMessageProvider.ERROR);
 			}
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/actions/ProfileModificationAction.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/actions/ProfileModificationAction.java
index 0773934..8bc179a 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/actions/ProfileModificationAction.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/actions/ProfileModificationAction.java
@@ -15,6 +15,7 @@ import java.lang.reflect.InvocationTargetException;
 import java.util.ArrayList;
 import java.util.List;
 import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
 import org.eclipse.equinox.internal.p2.ui.ProvUIMessages;
 import org.eclipse.equinox.internal.provisional.p2.core.ProvisionException;
 import org.eclipse.equinox.internal.provisional.p2.director.ProvisioningPlan;
@@ -104,6 +105,9 @@ public abstract class ProfileModificationAction extends ProvisioningAction {
 	 */
 	protected boolean validatePlan(ProvisioningPlan plan) {
 		if (plan != null) {
+			// Don't validate the plan if the user cancelled
+			if (plan.getStatus().getSeverity() == IStatus.CANCEL)
+				return false;
 			if (getPlanValidator() != null)
 				return getPlanValidator().continueWorkingWithPlan(plan, getShell());
 			if (plan.getStatus().isOK())
